services/user.py

```diff
@@ -0,0 +1,50 @@
+from django.contrib.auth import get_user_model
+from typing import Optional
+
+User = get_user_model()
+
+
+def create_user(
+    username: str,
+    password: str,
+    email: Optional[str] = None, first_name: Optional[str] = None,
+    last_name: Optional[str] = None
+) -> User:
+    user = User.objects.create_user(
+        username=username,
+        password=password)
+    if email:
+        user.email = email
+    if first_name:
+        user.first_name = first_name
+    if last_name:
+        user.last_name = last_name
+    user.save()
+    return user
+
+
+def get_user(user_id: int) -> Optional[User]:
+    return User.objects.get(id=user_id)
+
+
+def update_user(
+    user_id: int,
+    username: Optional[str] = None,
+    password: Optional[str] = None,
+    email: Optional[str] = None,
+    first_name: Optional[str] = None,
+    last_name: Optional[str] = None
+) -> User:
+    user = get_user(user_id)
+    if username:
+        user.username = username
+    if password:
+        user.set_password(password)
+    if email:
+        user.email = email
+    if first_name:
+        user.first_name = first_name
+    if last_name:
+        user.last_name = last_name
+    user.save()

```

Comment:
Just like in the `create_user()` function, you are calling `save()` method twice here: once in `set_password()` and once explicitly at line 49. To avoid this, you can use `check_password()` method to check if the new password is different from the old one, and only then call `set_password()` and `save()`. It's a minor performance improvement, but it's a good practice to avoid unnecessary database queries.